spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/discovery/KubernetesDiscoveryProperties.java

```diff
@@ -56,14 +58,16 @@ public record KubernetesDiscoveryProperties(
 		@DefaultValue Map<String, String> serviceLabels, String primaryPortName,
 		@DefaultValue Metadata metadata,
 		@DefaultValue("" + DEFAULT_ORDER) int order,
-		boolean useEndpointSlices) {
+		boolean useEndpointSlices,
+		boolean includeExternalNameServices) {
```

Comment:
add support to opt-in into this new functionality.

Reply:
Can we add a new constructor instead of modifying the existing one?

Reply:
I don't think we can for records, but we have defined for example `useEndpointSlices` in here in a previous PR

Reply:
I believe you can create multiple constructors for records.  I would prefer we just create a new one, its less disruptive, and this would be a breaking change.

Reply:
right, you can create multiple constructors, but at least one of them (that adds this new parameter `includeExternalNameServices`) will have to set an instance field and those are not allowed in records. I'll think about this more, but so far, have not seen an option to achieve this.

Reply:
Why wouldn't something like this work?

```
public record KubernetesDiscoveryProperties(
		@DefaultValue("true") boolean enabled, boolean allNamespaces,
		@DefaultValue Set<String> namespaces,
		@DefaultValue("true") boolean waitCacheReady,
		@DefaultValue("60") long cacheLoadingTimeoutSeconds,
		boolean includeNotReadyAddresses, String filter,
		@DefaultValue({"443", "8443"}) Set<Integer> knownSecurePorts,
		@DefaultValue Map<String, String> serviceLabels, String primaryPortName,
		@DefaultValue Metadata metadata,
		@DefaultValue("" + DEFAULT_ORDER) int order,
		boolean useEndpointSlices,
		boolean includeExternalNameServices) {

	public KubernetesDiscoveryProperties(@DefaultValue("true") boolean enabled, boolean allNamespaces,
		@DefaultValue Set<String> namespaces,
		@DefaultValue("true") boolean waitCacheReady,
		@DefaultValue("60") long cacheLoadingTimeoutSeconds,
		boolean includeNotReadyAddresses, String filter,
		@DefaultValue({"443", "8443"}) Set<Integer> knownSecurePorts,
		@DefaultValue Map<String, String> serviceLabels, String primaryPortName,
		@DefaultValue Metadata metadata,
		@DefaultValue("" + DEFAULT_ORDER) int order,
		boolean useEndpointSlices) {
		this(enabled, allNamespaces, namespaces, waitCacheReady, cacheLoadingTimeoutSeconds, includeNotReadyAddresses, filter, knownSecurePorts, serviceLabels, 
			primaryPortName, metadata, order, useEndpointSlices, false);
	}
....
```

Reply:
This was simply, sorry, I got confused a little here. I should have thought about this in reverse: add a new constructor to match the public one we already had. Thank you for the good tip. I also had to add:

```
	@ConstructorBinding
	public KubernetesDiscoveryProperties {

	}
```

otherwise the tests are failing because it seems that spring-boot is looking for a single constructor for final classes, so we have to tell it which one to use here.

Reply:
Great!  Shouldn't we no longer need all the constructor changes in the tests now?

Reply:
yeah, sorry, I don't get your point. :( Do you mean we should not longer enforce all those `@DefaultValue` via `KubernetesDiscoveryPropertiesTests`?

Reply:
No I mean where you added `false` in all the tests

Reply:
makes sense, done. seems like you can review again, your comments have been taken care of, imo.